### PR TITLE
Space management breadcrumb

### PR DESCRIPTION
**Pull request by 1egoman**
This pull request has the beginnings of the space management project contained within it.

- Adds a new page, routable at `#/admin/locations`. When loaded, it makes a request to the server to load all spaces (if all spaces are already loaded, the request is not made and the spaces that are already in the store are used instead)
- Adds a component that renders a breadcrumb relating to the currently selected space using the space's `ancestry` value.
- Currently, it renders a `<ul>` containing links to all direct children of the space that is currently selected. The interface that goes here is coming in a future pull request.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type DensitySpace = {
 	currentCount: number,
 	capacity: number,
 	createdAt: string,
+  ancestry: Array<DensitySpaceAncestryItem>,
```
Comment (guscost) [inline src/types.ts]: got some weird tab behavior here?

File: src/reducers/spaces/index.ts [hunk 2]
```diff
@@ -47,6 +48,7 @@ const DATA_DURATION_WEEK = 'DATA_DURATION_WEEK';
       // DATA_DURATION_MONTH = 'DATA_DURATION_MONTH';
 
 const initialState = {
+  view: 'LOADING',
```
Comment (guscost) [inline src/reducers/spaces/index.ts]: It bothers me that the dashboard uses `state` for this concept unlike almost everything else, and I didn't realize it until just now (when we probably shouldn't touch that stuff again for a while). This and other little inconsistencies with the collection action naming and such should all go in some sort of "fix these mildly annoying consistency pet peeves that gus complains about" project in our next engineering to-do list.

File: src/components/admin-locations/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            </AppBarSection>
+            <AppBarSection>
+              <span className={styles.leftButton}>
+                <Button disabled type="primary">
+                  <Skeleton width={96} color={colorVariables.grayLight} />
+                </Button>
+              </span>
+              <span className={styles.rightButton}>
+                <Button disabled type="primary">
+                  <Skeleton width={96} color={colorVariables.grayLight} />
```
Comment (guscost) [inline src/components/admin-locations/index.tsx]: ☠️ 👍

File: src/components/admin-locations-breadcrumb/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React, { Fragment } from 'react';
+import styles from './styles.module.scss';
+
+import { Icons } from '@density/ui';
+
+export default function Breadcrumb({ space }) {
```
Comment (guscost) [inline src/components/admin-locations-breadcrumb/index.tsx]: Shouldn't this be AdminLocationsBreadcrumb (another component name consistency grumbling goes here)

File: src/components/.template/index.tsx [hunk 5]
```diff
@@ -4,7 +4,9 @@ import styles from './styles.module.scss';
 export default function %COMPONENTUPPERCAMEL%({
   name,
 }) {
-  return <div className={styles.%COMPONENTCAMEL%}>
-    {name ? `Hello ${name}` : 'Hello World!'}
-  </div>;
+  return (
```
Comment (guscost) [inline src/components/.template/index.tsx]: Sure, whatever, let's waste a couple lines of code for nice clean parens 😛

File: src/actions/route-transition/admin-locations.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    dispatch({
+      type: ROUTE_TRANSITION_ADMIN_LOCATIONS,
+      parentSpaceId,
+      setLoading: shouldLoadSpaces,
+    });
+
+    if (shouldLoadSpaces) {
+      let spaces;
+      try {
+        const rawSpaces = await fetchAllPages(async page => {
```
Comment (guscost) [inline src/actions/route-transition/admin-locations.ts]: Should we abstract this chunk? It's used a lot and this is a nice version of it.

Comment (1egoman) [reply src/actions/route-transition/admin-locations.ts]: Haha I actually started going down the path to abstract this this morning! I had made an action called something like `collectionSpacesFetchAll` that had this logic in it and would be called from other actions. Let's talk about it and see how we can get that change in a separate PR.

File: src/components/admin-locations-breadcrumb/index.tsx (hunk 4 above)
Comment (1egoman) [reply src/components/admin-locations-breadcrumb/index.tsx]: Yep, will do

File: src/reducers/spaces/index.ts (hunk 2 above)
Comment (1egoman) [reply src/reducers/spaces/index.ts]: Sounds good, some consistency would be helpful!

File: src/actions/route-transition/admin-locations.ts (hunk 6 above)
Comment (guscost) [reply src/actions/route-transition/admin-locations.ts]: 👍

**Pull request merged**